client/lib/titan/new-mailbox.js

```diff
@@ -226,7 +259,46 @@ const transformMailboxForCart = ( {
 	password,
 } );
 
+const checkEmailAvailability = async ( emailAddress ) => {
+	try {
+		await wp.undocumented().getTitanEmailAddressAvailability( emailAddress );
+		return true;
+	} catch ( e ) {
+		return e.statusCode !== 409;
+	}
+};
+
+const decorateMailboxWithExistenceError = ( mailbox ) => {
+	mailbox.mailbox.error = translate( '{{strong}}%(email)s{{/strong}} already exists.', {
+		args: {
+			email: mailbox.mailbox.value,
+		},
+		components: {
+			strong: <strong />,
+		},
+	} );
+	return mailbox;
+};
+
+const areAllMailboxesAvailable = async ( mailboxes, onMailboxesChange ) => {
+	const promisified_checks = Promise.all(
+		mailboxes.map( ( mailbox ) => {
+			const email = `${ mailbox.mailbox.value }@${ mailbox.domain.value }`;
+			return checkEmailAvailability( email );
+		} )
+	);
+	const checks = await promisified_checks;
+	checks
+		.map( ( check, index ) => ( { check, mailbox: mailboxes[ index ] } ) )
+		.filter( ( { check } ) => ! check )
+		.forEach( ( { mailbox } ) => decorateMailboxWithExistenceError( mailbox ) );
+	const result = checks.every( ( check ) => check );
+	! result && onMailboxesChange && onMailboxesChange( mailboxes );
```

Comment:
Nit: I'd prefer this to be more explicit.
```suggestion
	if ( ! result && onMailboxesChange ) {
		onMailboxesChange( mailboxes );
	}
```